### ISSUE TITLE
Deriving Serialize for a struct from external crate which uses generics

### ISSUE THREAD

**Issue by AntoineBalaine** [serde-rs/serde#2327]
Hello, 
I'm trying to implement Serialize for [nom_locate's](https://docs.rs/nom_locate/latest/nom_locate/) [`LocatedSpan`](https://docs.rs/nom_locate/latest/nom_locate/struct.LocatedSpan.html). It's a struct that encapsulates a token with some location information.  can't figure out how to implement Serialize for it. 

I created a `WrappedSpan` to build the bridge with `LocatedSpan`, but the derive macro doesn't chew up the passed-in generic. Is there a way around this? 

```rust
#[derive(Serialize, Deserialize)]
#[serde(remote = "LocatedSpan<T>")] // <= This <T> produces an un-parseable token
struct WrappedSpan<T: AsBytes>{
    #[serde(getter = "LocatedSpan::location_offset")]
    offset: usize,
    #[serde(getter = "LocatedSpan::location_line")]
    line: u32,
    #[serde(getter = "LocatedSpan::fragment")]
    fragment: T,
};

// Provide a conversion to construct the remote type.
impl <T: AsBytes>From<WrappedSpan<T>> for LocatedSpan<T> {
    fn from(def: WrappedSpan<T>) -> LocatedSpan<T> {
        LocatedSpan{
            offset: def.offset,
            line: def.line,
            fragment: def.fragment,
            extra: (),
        }
    }
}
```

**Comment by AntoineBalaine** [comment]
Thank you for this update!

**Comment by dtolnay** [comment]
Fixed in serde_derive 1.0.148. You need to use `serde(remote = "LocatedSpan")`.

**Comment by AntoineBalaine** [comment]
Could you be a bit more specific ? I can't seem to figure it out with the update:
```rust
#[derive(Serialize, Deserialize, PartialEq, Eq)]
#[serde(remote = "LocatedSpan")]
struct WrappedSpan<T: AsBytes> {
    #[serde(getter = "LocatedSpan::location_offset")]
    offset: usize,
    #[serde(getter = "LocatedSpan::location_line")]
    line: u32,
    #[serde(getter = "LocatedSpan::fragment")]
    fragment: T,
}

// Provide a conversion to construct the remote type.
impl<T: AsBytes> From<WrappedSpan<T>> for LocatedSpan<T> {
    fn from(def: WrappedSpan<T>) -> LocatedSpan<T> {
        LocatedSpan::new(def.fragment)
    }
}

#[test]
fn serialize_located_span() {
    let input = LocatedSpan::new("hello");
    // Convert the Point to a JSON string.
    let serialized = serde_json::to_string(&input).unwrap(); //<= the trait bound `LocatedSpan<&str>: Serialize` is not satisfied
    let converted = "{\"offset\":0,\"line\":1,\"fragment\":\"\",\"column\":1}";
    assert_eq!(serialized, converted)
}
```
